From 54d328371656d4c8e8162f2d20210cab4d0bd322 Mon Sep 17 00:00:00 2001
From: Thang Nguyen <thang.nguyen.ue@renesas.com>
Date: Tue, 8 Mar 2022 14:38:38 +0700
Subject: [PATCH 3/8] Do not parse and init lossy mem area for RZ/V2N

RZ/V2N do not support lossy mem area.
So this commit is created to let mmngr not parse and init
lossy mem area for RZ/V2N.

Signed-off-by: Thang Nguyen <thang.nguyen.ue@renesas.com>
---
 .../mmngr-module/files/mmngr/drv/mmngr_drv.c  | 37 ++++++++++++-------
 1 file changed, 24 insertions(+), 13 deletions(-)

diff --git a/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c b/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c
index 5d31aa6..2eabed1 100644
--- a/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c
+++ b/mmngr_drv/mmngr/mmngr-module/files/mmngr/drv/mmngr_drv.c
@@ -1255,6 +1255,11 @@ static int parse_legacy_memory_node(struct device_node *np)
 }
 #endif
 
+static const struct soc_device_attribute rzv2n_match[] = {
+	{ .family = "RZ/V2N" },
+	{ /* sentinel*/ }
+};
+
 static int parse_reserved_mem_dt(struct device_node *np)
 {
 	int ret = 0;
@@ -1269,15 +1275,18 @@ static int parse_reserved_mem_dt(struct device_node *np)
 			 "(linux,multimedia) from DT\n");
 		return ret;
 	}
-	/* Parse reserved memory for lossy compression feature */
-	ret = _parse_reserved_mem_dt(np, "memory-region",
-				     "linux,lossy_decompress",
-				     &mm_lossybuf_addr,
-				     &mm_lossybuf_size);
-	if (ret) {
-		pr_warn("Failed to parse Lossy reserved area" \
-			"(linux,lossy_decompress) from DT\n");
-		ret = 0; /* Let MMNGR support other features */
+
+	if (!soc_device_match(rzv2n_match)) {
+		/* Parse reserved memory for lossy compression feature */
+		ret = _parse_reserved_mem_dt(np, "memory-region",
+					     "linux,lossy_decompress",
+					     &mm_lossybuf_addr,
+					     &mm_lossybuf_size);
+		if (ret) {
+			pr_warn("Failed to parse Lossy reserved area" \
+				"(linux,lossy_decompress) from DT\n");
+			ret = 0; /* Let MMNGR support other features */
+		}
 	}
 	return ret;
 }
@@ -1787,10 +1796,12 @@ static int mm_probe(struct platform_device *pdev)
 	}
 #endif
 
-	ret = init_lossy_info();
-	if (ret) {
-		pr_err("MMD mm_init ERROR\n");
-		return -1;
+	if (!soc_device_match(rzv2n_match)) {
+		ret = init_lossy_info();
+		if (ret) {
+			pr_err("MMD mm_init ERROR\n");
+			return -1;
+		}
 	}
 
 	p = kzalloc(sizeof(struct MM_DRVDATA), GFP_KERNEL);
-- 
2.17.1

